test/system/runScenario1.js

```diff
@@ -0,0 +1,13 @@
+const runScenario1 = (trader1) => {
+  return new Promise((resolve) => {
```

Comment:
Since trader1.placeOrder (and the chain that follows) returns a Promise, you can remove this line (and the .then(() => resolve(true)). Then runScenario1() can simply return the result of trader1.placeOrder('S').then().then()...catch();

There are a few other places where this can be done - I've marked with (*)